### PR TITLE
refactor: update 29-fee to use runtime.Environment

### PR DESCRIPTION
**Pull request by bznein**
## Description

ref: #7712 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/29-fee/keeper/keeper.go [hunk 1]
```diff
@@ -24,8 +22,8 @@ var _ types.ChannelKeeper = (*Keeper)(nil)
 
 // Keeper defines the IBC fungible transfer keeper
 type Keeper struct {
-	storeService corestore.KVStoreService
-	cdc          codec.BinaryCodec
+	appmodule.Environment
```
Comment (DimitrisJim) [inline modules/apps/29-fee/keeper/keeper.go:25]: Dittorino (as in other PR :sweat_smile:)

```suggestion
	appmodule.Environment

```

**Comment by gjermundgaraba** [comment]
@damiannolan I guess you can take over this one?

**Review by DimitrisJim — APPROVED** [review]

File: modules/apps/29-fee/keeper/grpc_query.go [hunk 2]
```diff
@@ -50,13 +50,11 @@ func (k Keeper) IncentivizedPackets(ctx context.Context, req *types.QueryIncenti
 }
 
 // IncentivizedPacket implements the Query/IncentivizedPacket gRPC method
-func (k Keeper) IncentivizedPacket(goCtx context.Context, req *types.QueryIncentivizedPacketRequest) (*types.QueryIncentivizedPacketResponse, error) {
+func (k Keeper) IncentivizedPacket(ctx context.Context, req *types.QueryIncentivizedPacketRequest) (*types.QueryIncentivizedPacketResponse, error) {
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
 
-	ctx := sdk.UnwrapSDKContext(goCtx).WithBlockHeight(int64(req.QueryHeight))
```
Comment (damiannolan) [inline modules/apps/29-fee/keeper/grpc_query.go:58]: Note that the understanding of `WithBlockHeight` when this was done by previous implementers was incorrect. This does not load the ibc store at the requested height only overwrite the context field at the given moment. 

At this point the multistore instance available to the request context has already been loaded by baseapp. The intended functionality for cosmos-sdk can be achieved by passing the `x-cosmos-block-height` grpc header via metadata

**Review by damiannolan — APPROVED** [review]
I will address events and branch service in a follow up PR as it was not done here.

**Pull request merged**